src/commands/mod.rs

```diff
@@ -275,6 +315,43 @@ where
     map.end()
 }
 
+fn deserialize_option_string_not_empty<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt: Option<String> = Option::deserialize(deserializer)?;
+
+    match opt {
+        None => Ok(None),
+        Some(ref s) if s.is_empty() => Err(serde::de::Error::custom("String cannot be empty")),
+        _ => Ok(opt),
+    }
+}
+
+
+fn serialize_setting_type<S>(setting_type: &SettingType, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let s = match setting_type {
+        SettingType::String => "string",
+        SettingType::Secret => "secret",
+    };
+    serializer.serialize_str(s)
+}
+
+fn deserialize_setting_type<'de, D>(deserializer: D) -> Result<SettingType, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    match s.to_lowercase().as_str() {
+        "string" => Ok(SettingType::String),
+        "secret" => Ok(SettingType::Secret),
+        _ => Err(serde::de::Error::custom("Field 'type_' must be either String or Secret")),
```

Comment:
it should be field type...

Reply:
Also please instead implement from str on the enum

Reply:
"Field type must be either String or Secret"?

Reply:
Field type should be one of the following:
- xxx
- xxx

And for that you ideally enumerate all enum values. It's ok if you don't do that programmatically. But it should be simple and won't require changes in any extra places if we add new setting type.

Reply:
This can be done with macros of some crate like https://docs.rs/strum/latest/strum/